content/library/advanced-features/dataframes.md

```diff
@@ -228,298 +224,88 @@ st.data_editor({
-
-```python
-ProgressColumn("Sales volume", min_value=0, max_value=1000, format="$%f")
-```
-
-</RefCard>
-
-</TileContainer>
-
-<Important>
-
-We will release in-depth documentation and a blog post on how to configure columns in the next two weeks. Keep at an eye out for updates on this page and the [Streamlit blog](https://blog.streamlit.io/).
-
-</Important >
-
-There are techniques you can use with Pandas today to render columns as checkboxes, selectboxes, and change the type of columns, that don't involve the column configuration API. We explore these techniques in the following sections.
-
-### Boolean columns (checkboxes)
-
-To render columns as checkboxes and clickable checkboxes in `st.dataframe` and `st.data_editor`, respectively, set the type of the Pandas column as `bool`.
-
-Here’s an example of creating a Pandas DataFrame with column `A` containing boolean values. When we display it using `st.dataframe`, the boolean values are rendered as checkboxes, where `True` and `False` values are checked and unchecked, respectively.
+You can use `st.data_editor` to collect tabular input from a user. When starting from an empty dataframe, default column types are text. Use column configuration to specify the data types you want to collect from users.
 
 ```python
+import streamlit as st
 import pandas as pd
 
-# create a dataframe with a boolean column
-df = pd.DataFrame({"A": [True, False, True, False]})
+df = pd.DataFrame(columns=['name','age','color'])
+colors = ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet']
+config = {
+    'name' : st.column_config.TextColumn('Full Name (required)', width='large', required=True),
+    'age' : st.column_config.NumberColumn('Age (years)', min_value=0, max_value=122),
+    'color' : st.column_config.SelectboxColumn('Favorite Color', options=colors)
+}
 
-# show the dataframe with checkboxes
-st.dataframe(df)
-```
-
-![data-editor-dataframe-boolean.gif](/images//data-editor-dataframe-boolean.gif)
-
-Notice you cannot change their values from the frontend. To let users check and uncheck values, we display the dataframe with `st.data_editor` instead:
+result = st.data_editor(df, column_config = config, num_rows='dynamic')
 
-```python
-import pandas as pd
-
-# create a dataframe with a boolean column
-df = pd.DataFrame({"A": [True, False, True, False]})
-
-# show the data editor with checkboxes
-st.data_editor(df)
+if st.button('Get results'):
+    st.write(result)
 ```
 
-![data-editor-boolean.gif](/images/data-editor-boolean.gif)
-
-### Categorical columns (selectboxes)
-
-To render columns as selectboxes with `st.data_editor`, set the type of the Pandas column as `category`:
-
-```python
-import pandas as pd
-
-df = pd.DataFrame(
-    {"command": ["st.selectbox", "st.slider", "st.balloons", "st.time_input"]}
-)
-df["command"] = df["command"].astype("category")
-
-edited_df = st.data_editor(df)
-```
+<Cloud src="https://doc-column-config-empty.streamlit.app/?embed=true" height="300"/>
```

Comment:
The current requirements file has an older version of pandas pinned. I wrote this app to use pandas>=2.0 since that's what installs by default for all versions of streamlit with column configuration. Can we unpin the version of pandas so it installs the latest? Or pin it to a higher version? Do we have any embedded examples relying on something deprecated in pandas?

Reply:
I have updated the pandas version to above 2.0.